conversions.js

```diff
@@ -100,48 +78,46 @@ convert.rgb.hsv = function (rgb) {
 	var b = rgb[2];
 	var min = Math.min(r, g, b);
 	var max = Math.max(r, g, b);
-	var delta = max - min;
-	var h;
-	var s;
-	var v;
+	var d = max - min;
+	var v = max / 255;
+	var s = max === 0 ? 0 : d / max;
 
-	if (max === 0) {
-		s = 0;
-	} else {
-		s = (delta / max * 1000) / 10;
+	if (d > 0) { // not achromatic
+		var h = max === r ? (g - b) / d + (g < b ? 6 : 0) :
+						max === g ? (b - r) / d + 2 :
+										(r - g) / d + 4;
+		return [(h * 60) % 360, s * 100, v * 100];
 	}
 
-	if (max === min) {
-		h = 0;
-	} else if (r === max) {
-		h = (g - b) / delta;
-	} else if (g === max) {
-		h = 2 + (b - r) / delta;
-	} else if (b === max) {
-		h = 4 + (r - g) / delta;
-	}
-
-	h = Math.min(h * 60, 360);
-
-	if (h < 0) {
-		h += 360;
-	}
-
-	v = ((max / 255) * 1000) / 10;
-
-	return [h, s, v];
+	return [0, s * 100, v * 100];
 };
 
 convert.rgb.hwb = function (rgb) {
-	var r = rgb[0];
-	var g = rgb[1];
-	var b = rgb[2];
-	var h = convert.rgb.hsl(rgb)[0];
-	var w = 1 / 255 * Math.min(r, Math.min(g, b));
-
-	b = 1 - 1 / 255 * Math.max(r, Math.max(g, b));
+	var r = rgb[0] / 255;
+	var g = rgb[1] / 255;
+	var b = rgb[2] / 255;
+	var min = Math.min(r, g, b);
+	var max = Math.max(r, g, b);
+	var d = max - min;
+	var black = 1 - max;
+
+	if (d > 0) { // not achromatic
+		var f;
+		var i;
+		if (min === r) {
+			f = g - b;
+			i = 180;
+		} else if (min === g) {
+			f = b - r;
+			i = 300;
+		} else {
+			f = r - g;
+			i = 60;
+		}
+		return [i - 60 * f / d, min * 100, black * 100];
+	}
 
-	return [h, w * 100, b * 100];
+	return [0, min * 100, black * 100];
```

Comment:
Hi Qix, I don't know if that's what you meant, but it could be simplified yes https://github.com/caub/color-convert/blob/c/conversions.js#L81-L99 

I used those sources for hwb https://github.com/caub/colorutil/blob/master/index.js#L5-L6 but the links are dead now

Reply:
There is now surely less branching, by avoiding convert.rgb.hsl